src/Gui/Placement.cpp

```diff
@@ -979,39 +849,307 @@ void Placement::changeEvent(QEvent *e)
+    hGrp->SetInt("RotationMethod", ui->rotationInput->currentIndex());
+
+    return true;
+}
+
+/*!
+ * \brief Placement::setSelection
+ * Sets the array of selection objects.
+ * \param selection
+ */
+void Placement::setSelection(const std::vector<Gui::SelectionObject>& selection)
+{
+    selectionObjects = selection;
+}
+
+void  Placement::setPropertyName(const std::string& name)
+{
+    handler.setPropertyName(name);
+}
+
+/*!
+ * \brief Placement::bindObject
+ * Binds the spin boxes to the placement components of the first object of the selection.
+ * This requires the call of \a setSelection() beforehand.
+ */
+void Placement::bindObject()
+{
+    if (!selectionObjects.empty()) {
+        App::DocumentObject* obj = selectionObjects.front().getObject();
+
+        std::string propertyName = handler.getPropertyName();
+        PlacementBase::bindObject(obj, propertyName);
+    }
+}
+// ----------------------------------------------------------------------------
+
+/* TRANSLATOR Gui::Dialog::PlacementDialog */
+
+PlacementDialog::PlacementDialog(App::DocumentObject* obj, std::string& pName, QWidget* parent, Qt::WindowFlags fl)
+    : PlacementBase(parent, fl)
+    , object(obj)
+    , propertyName(pName)
+{
+    ui->applyButton->hide();
+    ui->applyIncrementalPlacement->hide();
+
+    if (obj) {
+        auto* pPlc = dynamic_cast<App::PropertyPlacement*>(obj->getPropertyByName(propertyName.c_str()));
+        if (pPlc) {
+            backUpPlc = pPlc->getValue();
+        }
+        setPlacement(backUpPlc);
+        bindObject(obj, pName);
+    }
+    exec();
+}
+
+PlacementDialog::~PlacementDialog()
+{
+}
+
+void PlacementDialog::reject()
+{
+    applyPlacement(backUpPlc);
+    QDialog::reject();
+}
+
+void PlacementDialog::onPlacementChanged(int)
+{
+    Base::Placement plc = this->getPlacement();
+    applyPlacement(plc);
+}
+
+void PlacementDialog::applyPlacement(Base::Placement& plc)
+{
+    if (object) {
+        auto* pPlc = dynamic_cast<App::PropertyPlacement*>(object->getPropertyByName(propertyName.c_str()));
+        if (pPlc) {
+            pPlc->setValue(plc);
+        }
+    }
+}
+
+  /* Python binding */
+void PlacementDialogPy::init_type()
+{
+    behaviors().name("PlacementDialog");
+    behaviors().doc("PlacementDialog");
+    behaviors().set_tp_new(PyMake);
+    // you must have overwritten the virtual functions
+    behaviors().supportRepr();
+    behaviors().supportGetattr();
+    behaviors().supportSetattr();
+}
+
+PyObject* PlacementDialogPy::PyMake(struct _typeobject* type, PyObject* args, PyObject* kwds)
+{
+    Q_UNUSED(type)
+
+    PyObject* obj = nullptr;
+    const char* propName = nullptr;
+
+    if (!PyArg_ParseTuple(args, "O!s", &App::DocumentObjectPy::Type, &obj, &propName)) {
+        return nullptr;
+    }
+
+    auto docObj = static_cast<App::DocumentObjectPy*>(obj)->getDocumentObjectPtr();
+    std::string pName = propName;
+    return new PlacementDialogPy(new PlacementDialog(docObj, pName));

```

Comment:
PlacementDialog is owned by PlacementDialogPy. So, it is responsible to destroy it.